components/prism-agda.js

```diff
@@ -0,0 +1,29 @@
+(function (Prism) {
+
+	Prism.languages.agda = {
+		'comment': [
+      /\{\-[\s\S]*?(?:\-\}|$)/,
+      /\-\-.*/,
+    ],
+    'string': {
+      pattern: /"(?:\\(?:\r\n|[\s\S])|(?!")[^\\\r\n])*"/,
+      greedy: true,
+    },
+    'punctuation': /[(){}⦃⦄.;@]/,
+    'class-name': {
+      pattern: /((?:(?:data|record)) )[^\s{]+/,
+      lookbehind: true,
+    },
+    'function': {
+      pattern: /(\n\s*|^\s*)[^:\r\n]+?(?=:)/,
+      lookbehind: true,
+    },
+    'operator': {
+      pattern: /(^[(){}⦃⦄.;@\s]*|[(){}⦃⦄.;@\s]+)(?:[=|:∀→λ\\\?\_]|->)(?=[(){}⦃⦄.;@\s])/,
```

Comment:
All punctuation tokens (`[(){}⦃⦄.;@]`) have already been matched, so we can simplify this pattern.
```suggestion
      pattern: /(^\s*|\s)(?:[=|:∀→λ\\?_]|->)(?=\s)/,
```

Reply:
Thanks!